src/helpers.js

```diff
@@ -1,38 +1,68 @@
 'use babel'
 
-import ChildProcess from 'child_process'
-import { createFromProcess } from 'process-communication'
 import { join } from 'path'
 import escapeHTML from 'escape-html'
 import ruleURI from 'eslint-rule-documentation'
 import { generateRange } from 'atom-linter'
+import cryptoRandomString from 'crypto-random-string'
 
 // eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
-import { Disposable, Range } from 'atom'
+import { Range } from 'atom'
 
-export function spawnWorker() {
-  const env = Object.create(process.env)
-
-  delete env.NODE_PATH
-  delete env.NODE_ENV
-  delete env.OS
-
-  const child = ChildProcess.fork(join(__dirname, 'worker.js'), [], { env, silent: true })
-  const worker = createFromProcess(child)
+/**
+ * Start the worker process if it hasn't already been started
+ * @param  {Task} worker The worker process reference to act on
+ * @return {undefined}
+ */
+const startWorker = (worker) => {
+  if (worker.started) {
+    // Worker start request has already been sent
+    return
+  }
+  // Send empty arguments as we don't use them in the worker
+  worker.start([])
+  // NOTE: Modifies the Task of the worker, but it's the only clean way to track this
+  worker.started = true
+}
 
-  child.stdout.on('data', (chunk) => {
-    console.log('[Linter-ESLint] STDOUT', chunk.toString())
-  })
-  child.stderr.on('data', (chunk) => {
-    console.log('[Linter-ESLint] STDERR', chunk.toString())
-  })
+/**
+ * Send a job to the worker and return the results
+ * @param  {Task} worker The worker Task to use
+ * @param  {Object} config Configuration for the job to send to the worker
+ * @return {Object|String|Error}        The data returned from the worker
+ */
+export async function sendJob(worker, config) {
+  // Ensure the worker is started
+  try {
+    startWorker(worker)
+  } catch (e) {
+    throw e
```

Comment:
__*Question*__

What's the purpose of catching and re-throwing the error here (as well as in a few other places)?